safe_transaction_service/history/serializers.py

```diff
@@ -439,15 +398,56 @@ def validate(self, attrs):
                     f"Provided delegator={delegator} is not an owner of Safe={safe_address}"
                 )
 
-        # Tries to find a valid delegator using multiple strategies
-        for operation_hash in DelegateSignatureHelper.calculate_all_possible_hashes(
-            delegate
-        ):
-            if self.check_delegate_signature(
-                ethereum_client, signature, operation_hash, delegator
+    def validate_delegator_signature(
+        self,
+        delegate: ChecksumAddress,
+        signature: EthereumBytes,
+        signer: ChecksumAddress,
+    ) -> bool:
+        ethereum_client = EthereumClientProvider()
+        chain_id = ethereum_client.get_chain_id()
+        # Accept a message with the current topt and the previous totp (to prevent replay attacks)
+        for previous_totp in (True, False):
+            message_hash = DelegateSignatureHelper.calculate_hash(
+                delegate, chain_id, previous_totp=previous_totp
+            )
+            safe_signatures = SafeSignature.parse_signature(signature, message_hash)
+            if not safe_signatures:
+                raise ValidationError("Signature is not valid")
+
+            if len(safe_signatures) > 1:
+                raise ValidationError(
+                    "More than one signatures detected, just one is expected"
+                )
+            safe_signature = safe_signatures[0]
+            if safe_signature.signature_type not in (
+                SafeSignatureType.EOA,
+                SafeSignatureType.ETH_SIGN,
             ):
-                return attrs
+                raise ValidationError("Only EOA and ETH_SIGN signatures are supported")
```

Comment:
I couldn´t find this restriction in the previous version of serializer. Is this new restriction expected? 
If for example there is a Safe which one owner is other Safe, and this Safe delegated using contract signature EIPT1271 with the previous version wouldn´t be able to remove the delegate with the new version of the endpoint?

Reply:
As we didn't have the limit before, we shouldn't enforce now, good catch @moisses89